src/eredis.erl

```diff
@@ -118,6 +138,49 @@ q_async(Client, Command, Pid) when is_pid(Pid) ->
     Request = {request, create_multibulk(Command), Pid},
     gen_server:cast(Client, Request).
 
+%%--------------------------------------------------------------------
+%%% CREDENTIALS API
+%%--------------------------------------------------------------------
+
+-spec make_credentials(password() | credentials()) -> credentials().
+make_credentials(#{password := Password} = Credentials) ->
+    Username = maps:get(username, Credentials, undefined),
+    make_credentials(Username, Password);
+make_credentials(Password) ->
+    make_credentials(undefined, Password).
+
+-spec make_credentials(username() | undefined, password()) -> credentials().
+make_credentials(Username, Password) when
+    (Username =:= undefined orelse is_list(Username) orelse is_binary(Username)) andalso
+    (is_list(Password) orelse is_binary(Password) orelse is_function(Password, 0))
+->
+    #{
+        username => Username,
+        password => Password
+    }.
+
+-spec get_credentials_info(credentials()) -> #{
+        username := username() | undefined,
+        password := password()
+    }.
+get_credentials_info(#{username := Username, password := Password}) ->
+    #{
+        username => Username,
+        password => Password
+    }.
+
+-spec redact_credentials(credentials()) -> credentials().
+redact_credentials(Credentials) ->
+    Credentials#{
+        password => "******"
+    }.
+
+-spec make_empty_credentials() -> credentials().
+make_empty_credentials() ->
```

Comment:
Shouldn't this one include `username => undefined`?

Reply:
Ops... I forgot to change this on last patch.